processes/dags/pubmed_migration.py

```diff
@@ -111,59 +105,247 @@ def make_tarfile(output_filepath: str, source_dir: str) -> None:
 
     for filename in filenames:
-        ftp = FTP(ftp_url)
-        ftp.login(username, password)
-        ftp.cwd(ftp_path)
+        create_dir(temp_dir)
+        remote_path = os.path.join(ftp_path, filename)
+        local_path = os.path.join(temp_dir, filename)
+
+        ftp_hook.retrieve_file(remote_path, local_path)
+        o_path = extract_original_name(local_path)
+
+        extract_file_case_reports(local_path, o_path)
+        delete_file(local_path)
+        make_tarfile(local_path, o_path)
+        key = os.path.join(prefix, filename)
+        s3_hook.load_file(
+            local_path, key, bucket_name=bucket_name, replace=True)
+        delete_dir(temp_dir)
+
 
+def extract_pubmed_data_failure_callback(context) -> None:
+    delete_temp()
+
+
+def transform_pubmed_data() -> None:
+    """Downloads tarfile from S3, forms JSON files from contents, and uploads to S3
+    """
+    root_dir = '/usr/local/airflow'
+    temp_dir = os.path.join(root_dir, 'temp')
+    json_dir = os.path.join(temp_dir, 'json')
+    src_path = 'case_reports/pubmed/original/'
+    dest_path = 'case_reports/pubmed/json/'
+    src_bucket_name = 'supreme-acrobat-data'
+    dest_bucket_name = 'supreme-acrobat-data'
+    wildcard_key = 'case_reports/pubmed/original/*.*'
+    klist = s3_hook.list_keys(
+        src_bucket_name, prefix=src_path, delimiter='/')
+    key_matches = []
+    key_matches = [k for k in klist if fnmatch.fnmatch(k, wildcard_key)]
+
+    logging.info(key_matches)
+
+    for key in key_matches:
         create_dir(temp_dir)
-        filepath_tar_gz = download_file(ftp, filename, temp_dir)
-        filename_tar_gz = ntpath.basename(filepath_tar_gz)
-        o_path = extract_original_name(filepath_tar_gz)
-
-        extract_file(filepath_tar_gz, o_path)
-        delete_file(filepath_tar_gz)
-        make_tarfile(filepath_tar_gz, o_path)
-        upload_file_to_S3_with_hook(
-            filepath_tar_gz, filename_tar_gz, 'supreme-acrobat-data')
+        create_dir(json_dir)
+
+        basename = os.path.basename(key)
+        local_path = os.path.join(temp_dir, basename)
+        o_path = extract_original_name(local_path)
+        o_path_basename = os.path.basename(o_path)
+
+        obj = s3_hook.get_key(key, src_bucket_name)
+        obj.download_file(local_path)
+
+        extract_file(local_path, temp_dir)
+
+        glob_path = os.path.join(o_path, "*", "*.nxml")
+
+        filenames = [f for f in glob.glob(glob_path)]
+        logging.info("filenames = %s" % filenames)
+        json_list = []
+
+        for filename in filenames[0:4]:
+            fname, ext = os.path.splitext(os.path.basename(filename))
+            json_filename = ".".join((fname, "json"))
+            jl_filename = ".".join((fname, "jsonl"))
+            json_path = os.path.join(json_dir, json_filename)
+            jl_path = os.path.join(json_dir, jl_filename)
+            new_json = export_case_report_json(filename)
+            json_list.append(new_json)
```

Comment:
Some of these don't do anything. It might have been me who added some of it, but if you take over some code, you should remove unnecessary code.